plank/pkg/server/server_test_utils.go

```diff
@@ -0,0 +1,154 @@
+package server
+
+import (
+	"fmt"
+	"github.com/stretchr/testify/assert"
+	"github.com/vmware/transport-go/bus"
+	"github.com/vmware/transport-go/model"
+	"github.com/vmware/transport-go/plank/utils"
+	"io/ioutil"
+	"net"
+	"os"
+	"path/filepath"
+	"testing"
+	"time"
+)
+
+var testSuitePortMap = make(map[string]int)
+
+func getBasicTestServerConfig(rootDir, outLog, accessLog, errLog string, port int, noBanner bool) *PlatformServerConfig {
+	cfg := &PlatformServerConfig{
+		RootDir:                    rootDir,
+		Host:                       "localhost",
+		Port:                       port,
+		RestBridgeTimeoutInMinutes: time.Minute,
+		LogConfig: &utils.LogConfig{
+			OutputLog:     outLog,
+			AccessLog:     accessLog,
+			ErrorLog:      errLog,
+			FormatOptions: &utils.LogFormatOption{},
+		},
+		NoBanner:                 noBanner,
+		ShutdownTimeoutInMinutes: 1,
+	}
+	return cfg
+}
+
+func createTestServer(config *PlatformServerConfig) (baseUrl, logFile string, s PlatformServer) {
+	testServer := NewPlatformServer(config)
+
+	protocol := "http"
+	if config.TLSCertConfig != nil {
+		protocol += "s"
+	}
+
+	baseUrl = fmt.Sprintf("%s://%s:%d", protocol, config.Host, config.Port)
+	return baseUrl, config.LogConfig.OutputLog, testServer
+}
+
+func runWhenServerReady(t *testing.T, eventBus bus.EventBus, fn func(t *testing.T)) {
+	handler, _ := eventBus.ListenOnce(PLANK_SERVER_ONLINE_CHANNEL)
+	handler.Handle(func(message *model.Message) {
+		fn(t)
+	}, func(err error) {
+		assert.FailNow(t, err.Error())
+	})
+}
+
+func getTestPort() int {
```

Comment:
You should add this to the  `test_utils` directory and make it public.

Reply:
actually this causes an import cycle between `server` and `test_utils` packages because all the test cases are within the `server` package/folder and so it forms a circular reference of the two packages. i am afraid we might as well put the test harness stuff into the `server` package because i don't see a clean way to keep them in another package while preventing import cycles. let me know what you think

Reply:
OK, leave it as is. makes sence.